app/controllers/surveys_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SurveysController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def index
+    @surveys = Survey.all
+    @survey = Survey.new
+  end
+
+  def new
+    @survey = Survey.new
+    @question = @survey.questions.build
+    @option = @question.options.build
+  end
+
+  def show
+    @survey = Survey.find(params[:id])
+  end
+
+  def create
+    @survey = Survey.new(survey_params)
+    # @survey.questions.first.user_id = current_user.user_id
+    if @survey.save
+      redirect_to @survey
+    else
+      render action: :new
```

Comment:
what if any validation failed then user has to enter all the data again ?

Reply:
No the entered data remains in the text field